domain/application/service/application.go

```diff
@@ -132,6 +132,11 @@ type ApplicationState interface {
 	// application doesn't exist.
 	AddUnits(ctx context.Context, applicationName string, args ...application.UpsertUnitArg) error
 
+	// GetUnitName returns the name of the unit with the specified UUID, returning
+	// an error satisfying [applicationerrors.UnitNotFound] if the unit doesn't
```

Comment:
Are we providing the errors it might return, which is an implementation concern, on the interface's doc? I know you didn't start with this pattern on this PR, but if we want to go down this road (and I prefer not to) we should at least homogenize.

Reply:
Yeah it seems so. It isn't just an implementation concern since downstream may want to branch on error type